tests/jeos/prepare_firstboot.pm

```diff
@@ -47,6 +47,8 @@ sub run {
     if ($is_generalhw_via_ssh) {
         # Do not set eth0 down as we are connected through ssh!
         assert_script_run("sed -i 's/ip link set down \"\$d\" #/if [ \"eth0\" != \"\$d\" ]; then ip link set down \"\$d\"; fi; #/' /usr/lib/jeos-firstboot");
+        # 2nd version of the script, dpending on 'jeos-firstboot' version
+        assert_script_run("sed -i 's/ip link set down \"\$net_device\" #/if [ \"eth0\" != \"\$net_device\" ]; then ip link set down \"\$net_device\"; fi; #/' /usr/lib/jeos-firstboot");
     }
```

Comment:
Should verify that one of the replacements was successful, by either grepping for `if [ \"eth0\" != \"\$net_device\" ];` or using the `sed 's/i/a/;tg;q1;:g'` method.

Reply:
It can be `eth0` or `enp0s3` or maybe something else.

Reply:
No, because it can be `$d`or `$net_device`.

Reply:
Well, the pattern would just have to be universal. The `sed` exit code method would still work as well.

Reply:
Not sure how to make it universal, without commenting out `ip link set down`.

Reply:
I meant the pattern for checking, but this works too. Is it bad if it continues without having it commented out properly? It would probably fail anyway?

Reply:
Yes, it will fail quickly since you shutdown the network and thus ssh will break quickly.

Reply:
But other than that there are no worse consequences, right? In that case, it wouldn't be necessary to absolutely ensure that `ip link set down` got commented out.

Reply:
I cannot see any worse consequence. ;)